### PR TITLE
[24.0] Fix tool version switch in editor

### PR DESCRIPTION
**Pull request by mvdbeek**
Fixes https://github.com/galaxyproject/galaxy/issues/17464

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Review by jmchilton — APPROVED** [review]

File: lib/galaxy/managers/workflows.py [hunk 1]
```diff
@@ -1472,7 +1478,7 @@ def _workflow_to_dict_export(self, trans, stored=None, workflow=None, internal=F
                 "tool_version": module.get_version() if allow_upgrade else step.tool_version,
                 "name": module.get_name(),
                 "tool_state": json.dumps(tool_state),
-                "errors": module.get_errors(),
+                "errors": errors or module.get_errors(),
```
Comment (nsoranzo) [inline lib/galaxy/managers/workflows.py:1481]: I suppose a user would like to see the errors from `module.get_errors()` also when `errors` is not empty, i.e. we could merge them?
But currently `get_errors()` can return `None`, a list or a string, depending on the module type, which is not great in itself, so I suppose we need to decide what format this should have.

Comment (mvdbeek) [reply lib/galaxy/managers/workflows.py:1481]: errors will contain the upgrade messages, I don't think there's a need to merge them ?

Comment (nsoranzo) [reply lib/galaxy/managers/workflows.py:1481]: Then maybe put them in a "messages" key?

Comment (mvdbeek) [reply lib/galaxy/managers/workflows.py:1481]: Then I have to redo the editor and potentially other things. I don't disagree that this should be modernized, but it doesn't seem like the right moment ?

Comment (mvdbeek) [reply lib/galaxy/managers/workflows.py:1481]: And sorry, it's not messages, these are actually errors, for the current state of the module.

Comment (nsoranzo) [reply lib/galaxy/managers/workflows.py:1481]: Does the editor work with something that can be None, list, string or dictionary?
Anyway, my main concern was about losing the errors from `module.get_errors()` when `errors` is not empty.

Comment (mvdbeek) [reply lib/galaxy/managers/workflows.py:1481]: Yes, the editor just dumps them as a string, so anything's fine. I don't think we really ever display them for more than a second inside the node body, and the refactoring action just uses step.upgrade_messages.

**Pull request merged**